src/entities/async-exports/async-export.ts

```diff
@@ -0,0 +1,259 @@
+  public deleted?: AsyncExportPayload['deleted']
+  public active?: AsyncExportPayload['active']
+
+  public jobReferenceId?: AsyncExportPayload['jobReferenceId']
+  public fileId?: AsyncExportPayload['fileId']
+  public signedAt?: AsyncExportPayload['signedAt']
+  public expiresAt?: AsyncExportPayload['expiresAt']
+
+  public type?: AsyncExportPayload['type']
+  public request?: AsyncExportPayload['request']
+  public staffIds?: AsyncExportPayload['staffIds']
+
+  constructor (options: AsyncExportOptions) {
+    super()
+    this.universe = options.universe
+    this.apiCarrier = options.universe
+    this.endpoint = 'api/v1/exports'
+    this.http = options.http
+    this.options = options
+    this.initialized = options.initialized ?? false
+
+    if (options?.rawPayload) {
+      this.deserialize(options.rawPayload)
+    }
+  }
+
+  protected deserialize (rawPayload: AsyncExportRawPayload): this {
+    this.setRawPayload(rawPayload)
+
+    this.type = rawPayload.type
+    this.request = rawPayload.request
+    this.staffIds = rawPayload.staff_ids
+
+    return this
+  }
+
+  public static create (payload: AsyncExportRawPayload, universe: Universe, http: Universe['http']): AsyncExport {
+    return new AsyncExport({ rawPayload: payload, universe, http, initialized: true })
+  }
+
+  public serialize (): AsyncExportRawPayload {
+    return {
+      id: this.id,
+      created_at: this.createdAt ? this.createdAt.toISOString() : undefined,
+      updated_at: this.updatedAt ? this.updatedAt.toISOString() : undefined,
+      deleted: this.deleted,
+      active: this.active,
+
+      job_reference_id: this.jobReferenceId,
+      file_id: this.fileId,
+      signed_at: this.signedAt,
+      expires_at: this.expiresAt,
+
+      type: this.type,
+      // What will this look like
+      request: this.request,
+      staff_ids: this.staffIds
+    }
+  }
+
+  public async init (): Promise<AsyncExport | undefined> {
+    try {
+      await this.fetch()
+
+      return this
+    } catch (err) {
+      throw this.handleError(new AsyncExportInitializationError(undefined, { error: err }))
+    }
+  }
+
+  // This schedules a job for exporting data. The job will be executed asynchronously.
+  // We only care about the status code here, so we don't need to return the payload.
+  public async createExport (payload: AsyncExportRawPayload | AsyncExportPayload): Promise<AxiosResponse> {
+    try {
+      const opts = {
+        method: 'POST',
+        url: `${this.universe.universeBase}/${this.endpoint}`,
+        headers: {
+          'Content-Type': 'application/json; charset=utf-8'
+        },
+        data: payload,
+        responseType: 'json'
+      }
+
+      const response = await this.http.getClient()(opts)
+
+      // Maybe not needed but for now
+      // const responseData = response.data.data[0]
+
+      return response
+    } catch (err) {
+      throw this.handleError(new AsyncExportInitializationError(undefined, { error: err }))
```

Comment:
Todo: We should also handle the case when we return the upsert response with its dedicated message/status code.

Reply:
I think looking at the [BE](https://github.com/c-commerce/charles-client-api/commit/28326ed566c6fd52f37ae004a8cbebaef8b4cee6#diff-bfc5298fae14ad5976abce33067a194fe517f0af5d87dfdd466627b72e4d33a5R262-R296)

We already handle the status codes so the response should just concern itself with which one and pass this on with the message, and I guess the Raw Payload incase we need that at somepoint